src/kml/src/main/java/org/geoserver/kml/utils/KmlCentroidBuilder.java

```diff
@@ -26,6 +36,38 @@
      * </p>
      */
     public Coordinate geometryCentroid(Geometry g) {
+        return geometryCentroid(g, null,null);
+    }
+
+    /**
+     * Returns the centroid of the geometry, handling a geometry collection.
+     * <p>
+     * In the case of a collection a multi point containing the centroid of each geometry in the
+     * collection is calculated. The first point in the multi point is returned as the controid.
+     * </p>
+     * <p>
+     * The <tt>opts</tt> parameter is used to provide additional options controlling how the 
+     * centroid is computed.
+     * </p>
+     * @param g The geometry to compute the centroid.
+     * @param bbox The request bbox, used to potentially clip the geometry before computting the centroid.
+     * @param opts The centroid options controlling whether clipping/sampling/etc... are used.
+     */
+    public Coordinate geometryCentroid(Geometry g, Envelope bbox, KmlCentroidOptions opts) {
+        if (opts == null) {
+            opts = KmlCentroidOptions.DEFAULT;
+        }
+
+        // clip?
+        if (opts.isClip()) {
+            if (bbox != null) {
+                g = clipGeometry(g, bbox);
+            }
```

Comment:
Else should be on the same line as closing curly brace. Our conventions are the Java ones minus a few amendments. See: http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html